carla_integration/carma_docker.launch.py

```diff
@@ -30,9 +31,18 @@ def generate_launch_description():
     config_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'carma_rosconsole.conf') 
     logging_env_var = SetEnvironmentVariable('CARMA_ROS_LOGGING_CONFIG', generate_log_levels(config_file_path))
 
+    # Obtain the vehicle calibration directory
+    vehicle_calibration_dir = '/opt/carma/vehicle/calibration'
+    DeclareLaunchArgument(
+        name = 'vehicle_calibration_dir', 
+        default_value = vehicle_calibration_dir, 
+        description = "Path to vehicle calibration directory"
+    )
+
     # Launch the core carma launch file
     carma_src_launch = IncludeLaunchDescription(
-        PythonLaunchDescriptionSource([ get_package_share_directory('carma'), '/launch/carma_src.launch.py'])
+        PythonLaunchDescriptionSource([ get_package_share_directory('carma'), '/launch/carma_src.launch.py']),
+        launch_arguments = { 'vehicle_calibration_dir' : vehicle_calibration_dir }.items()
```

Comment:
This launch argument will not actually get set the way you expect since vehicle_calibration_dir is a variable and not a launch substitution it will only ever hold its initial value. 
In order to actually access the argument you need a LaunchConfiguration object. For example this is what the template package does. The name parameter of the DeclareLaunchArgument and the LaunchConfiguration must be the same for the substitution to resolve. The log_level object would then be what you could pass as launch_arguments to other files. 
```
log_level = LaunchConfiguration('log_level')
    declare_log_level_arg = DeclareLaunchArgument(
        name ='log_level', default_value='WARN')
 ```
 
 Additionally, they must be included in the returned LaunchDescription at the bottom of this file before any components that use them.